multimodal/src/autogluon/multimodal/data/process_document.py

```diff
@@ -0,0 +1,331 @@
+import logging
+import os
+import warnings
+from io import BytesIO
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+import PIL
+import pytesseract
+import torch
+from nptyping import NDArray
+from PIL import ImageFile
+from torch import nn
+from torchvision import transforms
+from transformers import AutoTokenizer
+
+from ..constants import AUTOMM, BBOX
+from .collator import PadCollator, StackCollator
+from .utils import construct_processor, mean_std
+
+logger = logging.getLogger(AUTOMM)
+
+# Disable tokenizer parallelism
+os.environ["TOKENIZERS_PARALLELISM"] = "false"
+
+
+class DocumentProcessor:
+    """
+    Prepare document data for Document Classification.
+    OCR (Optical character recognition) is applied to get the document texts and bounding boxes.
+    Both texts and images will be processed.
+    """
+
+    def __init__(
+        self,
+        model: nn.Module,
+        train_transform_types: List[str],
+        val_transform_types: List[str],
+        norm_type: Optional[str] = None,
+        size: Optional[int] = None,
+        text_max_len: Optional[int] = 512,
+    ):
+        """
+        Parameters
+        ----------
+        model
+            The model using this data processor.
+        train_transform_types
+            A list of image transforms used in training. Note that the transform order matters.
+        val_transform_types
+            A list of image transforms used in validation/test/prediction. Note that the transform order matters.
+        norm_type
+            How to normalize an image. We now support:
+            - inception
+                Normalize image by IMAGENET_INCEPTION_MEAN and IMAGENET_INCEPTION_STD from timm
+            - imagenet
+                Normalize image by IMAGENET_DEFAULT_MEAN and IMAGENET_DEFAULT_STD from timm
+            - clip
+                Normalize image by mean (0.48145466, 0.4578275, 0.40821073) and
+                std (0.26862954, 0.26130258, 0.27577711), used for CLIP.
+        size
+            The width / height of a square image.
+        text_max_len
+            The max text length of tokenizer. Default: 512.
+        """
+        self.prefix = model.prefix
+        self.model = model
```

Comment:
Consider not setting model as an attribute of processor. We can just unpack the input keys as the attribute instead.